src/python/platforms/fuchsia/device.py

```
@@ -54,98 +53,140 @@ def setup_qemu_values(initial_setup=True):
+        '-machine', 'q35',
+        '-display', 'none',
+        '-netdev',
+        ('user,id=net0,net=192.168.3.0/24,dhcpstart=192.168.3.9,'
+         'host=192.168.3.2,hostfwd=tcp::') + str(port) + '-:22',
+        '-device', 'e1000,netdev=net0,mac=52:54:00:63:5e:7b',
+        '-L', qemu_vars['sharefiles_path']
+    ]
+    # yapf: enable
+
+    # Detecing KVM is tricky, so use an environment variable to determine
+    # whether to turn it on or not.
+    if environment.get_value('FUCHSIA_USE_KVM'):
+      qemu_args.extend(['-cpu', 'host,migratable=no'])
+      qemu_args.append('-enable-kvm')
+    else:
+      # Can't use host CPU since we don't necessarily have KVM on the machine.
+      # Emulate a Haswell CPU with a few feature toggles. This mirrors the most
+      # common configuration for Fuchsia VMs when using in-tree tools.
+      qemu_args.extend(['-cpu', 'Haswell,+smap,-check,-fsgsbase'])
+
+    # Get the list of fuzzers for ClusterFuzz to choose from.
+    host = Host.from_dir(
+        os.path.join(qemu_vars['fuchsia_resources_dir'], 'build', 'out',
+                     'default'))
+    Device(host, 'localhost', str(port))
+    Fuzzer.filter(host.fuzzers, '')
+
+    # Fuzzing jobs that SSH into the QEMU VM need access to this env var.
+    environment.set_value('FUCHSIA_PKEY_PATH', qemu_vars['pkey_path'])
+    logs.log('Ready to run QEMU. Command: ' + qemu_vars['qemu_path'] + ' ' +
+             str(qemu_args))
+    self.process_runner = new_process.ProcessRunner(qemu_vars['qemu_path'],
+                                                    qemu_args)
+
+  def run(self):
+    """Actually runs a QEMU VM, assuming `create` has already been called."""
+    if not self.process_runner:
+      raise QemuException('Attempted to `run` QEMU VM before calling `create`')
+    self.popen = self.process_runner.run(
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    time.sleep(_QEMU_WAIT_SECONDS)
+
+  def kill(self):
+    if not self.popen:
```

nit: docstring. also maybe raise an exception instead?

---

added a docstring.

presumably if someone's calling "kill", they just want the process good and dead, and we shouldn't throw an error if it happens to already be dead?  (i notice terminate_processes_matching_names just ignores psutil.NoSuchProcess errors, for example)

---

sure, that seems reasonable.